src/coral_models/prepare_raw_data.py

```diff
@@ -0,0 +1,470 @@
+        correct_country_list
+    )
+
+    # Specification of gender is not consistent, so we need to correct that
+    # by mapping "K" to "female", and "M" to "male".
+    speakers["gender"] = speakers["gender"].apply(
+        lambda x: "male"
+        if x in ["male", "M"]
+        else "female"
+        if x in ["K", "female"]
+        else x
+    )
+
+    # Create a speaker id column
+    speakers["speaker_id"] = "t" + speakers.index.astype(str)
+
+    return speakers
+
+
+def make_recording_metadata(
+    cfg: DictConfig, speakers: pd.DataFrame, raw_path: Path, sentences: pd.DataFrame
+) -> pd.DataFrame:
+    """Make a recording metadata file from the raw data
+
+    Args:
+        cfg (DictConfig):
+            The config
+        speakers (pd.DataFrame):
+            The speakers dataframe
+        raw_path (Path):
+            The path to the raw data
+        sentences (pd.DataFrame):
+            The sentences dataframe
+
+    Returns:
+        pd.DataFrame: The recording metadata dataframe"""
+    # Load the metadata
+    metadata_path = raw_path / "metadata.xlsx"
+
+    # Make speaker-metadata file
+    speaker_metadata = pd.read_excel(metadata_path)
+
+    # Get the columns that contain information about speakers, recorders and recording
+    recording_metadata = speaker_metadata[
+        [
+            col
+            for col in speaker_metadata.columns
+            if not any(
+                [
+                    substring in col
+                    for substring in ["subject_a", "subject_b", "recorder"]
+                ]
+            )
+        ]
+        + ["subject_a_mail", "subject_b_mail", "recorder_mail"]
+    ]
+
+    # Make a dict from speaker mail to speaker id in the speakers dataframe
+    speaker_mail_to_id = dict(zip(speakers["mail"], speakers["speaker_id"]))
+
+    # Make a speaker_id column for the recording metadata, with ids separated by a comma
+    # Subject A is the first speaker id, subject B is the second speaker id
+    recording_metadata["speaker_id"] = recording_metadata[
+        ["subject_a_mail", "subject_b_mail"]
+    ].apply(lambda x: ",".join([str(speaker_mail_to_id[mail]) for mail in x]), axis=1)
+
+    # Make a recorder_id column for the recording metadata
+    recording_metadata["recorder_id"] = recording_metadata["recorder_mail"].apply(
+        lambda x: str(speaker_mail_to_id[x])
+    )
+
+    # Drop columns
+    recording_metadata = recording_metadata.drop(
+        columns=["subject_a_mail", "subject_b_mail", "recorder_mail"]
+    )
+
+    # Make a sentence_id column for the recording metadata
+    recording_metadata["sentence_id"] = -1
+
+    # Make a sentence content to sentence_id dict
+    sentence_content_to_id = dict(zip(sentences["text"], sentences["sentence_id"]))
+
+    # Load speaker information from read aloud data
+    recording_metadata_list = [recording_metadata]
+    read_aloud_paths = raw_path.glob("*_oplæst_*")
```

Comment:
It seems like there's a decent amount of duplicate code in these two functions. Couldn't a lot of the functionality be moved to smaller functions, that are called in both "major" functions?